### PR TITLE
feat(angular) : Adding phone number field in angular primitive poc.

### PR DESCRIPTION
**Pull request by SantoshLatpate**
#### Description of changes
Adding development of phone number field.
Adding examples for phone number field.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    // this can be implemented later for the purpose of POC.
+    this.childInput.hasError = this.hasError;
+    this.childInput.ariaDescribedBy = this.describedBy;
+  }
+  onCodeChange(value) {
+    this.countryCodeChange.emit(value);
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    console.log('valuy ', this.defaultCountryCode);
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts:49]:
```suggestion
```

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  EventEmitter,
+  Input,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+import { countryDialCodes } from '@aws-amplify/ui';
+import { nanoid } from 'nanoid';
+import { PhoneNumberDirective } from '../phone-number-directive/phone-number.directive';
+
+const $COMPONENT_SELECTOR: string = 'amplify-password-field';
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts:14]: Wrong component selector here

File: examples/angular/src/pages/ui/primitive-poc/phone-number/phone-number.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<amplify-password-field
```
Comment (wlee221) [inline examples/angular/src/pages/ui/primitive-poc/phone-number/phone-number.component.html:1]: These should be phone number fields

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts (hunk 2 above)
Comment (wlee221) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts:14]: `amplify-phone-number-field-poc`?

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-directive/phone-number.directive.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { Directive, HostBinding, Input } from '@angular/core';
+import { nanoid } from 'nanoid';
+
+const $Directive_Selector = '[amplify-phone-number]';
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-directive/phone-number.directive.ts:4]: maybe `amplify-phone-number-input`. Later, we should abstract these out into a single `[amplify-input]` directive to remove duplciation.

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number.module.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { AmplifyPhoneNumberFieldComponent } from './phone-number-component/phone-number.component';
+import { PhoneNumberDirective } from './phone-number-directive/phone-number.directive';
+
+let Components: any[] = [
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number.module.ts:6]: Do you know if we can avoid `any[]` here?

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts (hunk 1 above)
Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts:49]: Removed.

Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts:49]: [075752a](https://github.com/aws-amplify/amplify-ui/pull/2420/commits/075752a0a6eb028fab66163ccc05ff6c0e625e94)

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts (hunk 2 above)
Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts]: Changed.

Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-component/phone-number.component.ts]: [075752a](https://github.com/aws-amplify/amplify-ui/pull/2420/commits/075752a0a6eb028fab66163ccc05ff6c0e625e94)

File: examples/angular/src/pages/ui/primitive-poc/phone-number/phone-number.component.html (hunk 3 above)
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/phone-number/phone-number.component.html]: Done.

Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/phone-number/phone-number.component.html]: [075752a](https://github.com/aws-amplify/amplify-ui/pull/2420/commits/075752a0a6eb028fab66163ccc05ff6c0e625e94)

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-directive/phone-number.directive.ts (hunk 4 above)
Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-directive/phone-number.directive.ts]: Changed.

Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number-directive/phone-number.directive.ts]: [075752a](https://github.com/aws-amplify/amplify-ui/pull/2420/commits/075752a0a6eb028fab66163ccc05ff6c0e625e94)

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number.module.ts (hunk 5 above)
Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number.module.ts]: We make it const and directly assign the array, Its constant so no need to give any[].

Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/phone-number/phone-number.module.ts]: [075752a](https://github.com/aws-amplify/amplify-ui/pull/2420/commits/075752a0a6eb028fab66163ccc05ff6c0e625e94)

**Review by wlee221 — APPROVED** [review]
Great work!

**Pull request merged**